File: it/ext/e04/examples/src/lesson4/Cash.java
package lesson4;

interface Money {
  Money multiply(float factor);
  String balance();
}

public class Cash implements Money {
  private final float dollars;
  @Override
  public Cash multiply(float factor) { return new Cash(this.dollars * factor); }
  @Override
  public String balance() { return "$" + dollars; }
  public Cash(float dollars) { this.dollars = dollars; }

  public static void main(String[] args) {
    Cash smallAmount = new Cash(15.0f);
    Cash bigAmount = smallAmount.multiply(10f);
    System.out.println("I have " + bigAmount.balance());
  }
}

File: code-examples/cpp/05-immutability-interfaces-and-proj-struct/src/Cash.cpp
#include "../include/Cash.h"

std::string Cash::Balance() {
  std::stringstream stream;
  stream << std::fixed << std::setprecision(2) << dollars;
  std::string s = stream.str();
  return "$" + s;
}
Cash *Cash::MultipliedBalance(float factor) {
  Cash *b = new Cash(dollars * factor);
  return b;
}
Cash::Cash(float dollars) {
  this->dollars = dollars;
}
